### ISSUE TITLE
search query string

### ISSUE THREAD

**Issue by incanus** [SwiftDocOrg/swiftdoc.org#4]
I would love to be able to hook this up to, say, [LaunchBar](http://www.obdev.at/products/launchbar/index.html) with a query string argument, like so: 

    http://swiftdoc.org/?q=map

And thus get either: 

1. A landing page showing search results for `map` across functions, types, etc. 

2. Combined with #3, get actual results in LaunchBar that I can select from. 

See for example: 

http://en.cppreference.com/mwiki/index.php?title=Special%3ASearch&search=std::map

**Comment by natecook1000** [comment]
I think the API will be great for this use case as well -- the site is completely static (built via Jekyll) so it would be tricky to support that kind of query.

In the mean time, you can use Google to search the site directly with this: https://www.google.com/search?q=site:swiftdoc.org+map

**Comment by natecook1000** [comment]
And of course, right now the site search is fairly limited in that it only looks up type and function names, not methods of a type or text in the comments. That could handle some improvement.

**Comment by incanus** [comment]
One way I've accomplished this in the past with Jekyll is to generate a JSON file with the relevant data and perform a search in-browser with JavaScript. See for example http://www.developmentseed.org and try searching for e.g. `application`. This is a flat-file Jekyll site.

**Comment by natecook1000** [comment]
Right, that's more or less what the search box is doing now, though its scope could be significantly expanded. I guess I was getting hung up on your second bullet point - does LaunchBar do more than fetching the HTML of a page? Javascript-based search results wouldn't be visible in that case.

**Comment by incanus** [comment]
>  does LaunchBar do more than fetching the HTML of a page

Yes, it can interact in a number of languages with e.g. JSON data. 

![anigif-1421335360](https://cloud.githubusercontent.com/assets/17722/5760126/6393476c-9c87-11e4-9690-af14a5cfeb2f.gif)

**Comment by natecook1000** [comment]
@incanus Very early API up here -- let me know your thoughts:

http://api.swiftdoc.org/

**Comment by incanus** [comment]
Nice, this looks pretty good. I will try to give it a spin. One question — why the duplicate entries? See for example http://api.swiftdoc.org/search?q=map.

**Comment by natecook1000** [comment]
It's giving you an entry for each global map function - there are three that each have slightly different generic constraints, but they're all on the same page. 

Let me know how it goes - I'll be putting the API itself up on GitHub soon, so we can discuss it further there.

**Comment by natecook1000** [comment]
In case anyone's looking, this now lives in [its own repo](https://github.com/SwiftDocOrg/api.swiftdoc.org) and is hosted on Heroku at [api.swiftdoc.org](http://api.swiftdoc.org/).
